### PR TITLE
[RPC] Update Sapling HD chain seed with sethdseed

### PR DESCRIPTION
**Pull request by random-zebra**
Currently `sethdseed` RPC only sets the seed for the transparent addresses HD chain.
Set it for the sspkm as well.
Bonus: add shield addresses support to `getaddressinfo`.

Update `wallet_hd.py` functional test, checking derivation of shield addresses too.

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -198,20 +203,37 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
...
+            auto it = sspk_man->mapSaplingZKeyMetadata.find(ivk);
+            if (it != sspk_man->mapSaplingZKeyMetadata.end()) {
+                meta = it->second;
+            }
+        }
+    }
+
+    // Add metadata
+    ret.pushKV("timestamp", meta.nCreateTime);
+    if (meta.HasKeyOrigin()) {
```
Comment (furszy) [inline src/wallet/rpcwallet.cpp:233]: The metadata could still be null here if there is no metadata for the address inside the wallet, and be loaded with the default constructor initialization (line 202). 
To keep the same flow as before, and not return the default values, should move it to a pointer and skip it if it's null.

Comment (random-zebra) [reply src/wallet/rpcwallet.cpp:233]: done

**Review by furszy — APPROVED** [review]
cool, code review ACK 092f82516bf6ba722d73d653a7010adcec211d5c.

**Review by Fuzzbawls — APPROVED** [review]
utACK 092f82516bf6ba722d73d653a7010adcec211d5c

**Pull request merged**